entities/src/main/java/org/odk/collect/entities/javarosa/parse/EntityFormParseProcessor.java

```diff
@@ -54,11 +56,11 @@ public void processBindAttribute(String name, String value, DataBinding binding)
 
     @Override
     public void processFormDef(FormDef formDef) throws XFormParser.ParseException {
-        if (!versionPresent && EntityFormParser.getEntityElement(formDef.getMainInstance()) != null) {
+        if (version == null && EntityFormParser.getEntityElement(formDef.getMainInstance()) != null) {
```

Comment:
Why do we need to check if `EntityFormParser.getEntityElement(formDef.getMainInstance()) != null` here?
Can't we just throw that exception if version is null?

Reply:
Hmm good point! Let me play with that.

Reply:
Ah, because if we do that, we end up exploding when there's no version and no entity element. We only care about validating the version if there's an entity element.

Reply:
Ok so it determines if a form contains entities or not? If so please create a method with a descriptive name that could be used here instead of that enigmatic 
`EntityFormParser.getEntityElement(formDef.getMainInstance()) != null`

Reply:
Gotha! I can probably wrap the whole method in that check as well to make it clearer.